adhoc-openshift-pv.yml

```diff
@@ -0,0 +1,57 @@
+---
+# This playbook is to create a persistent volume for openshift CI namespace
+# Example to run the playbook
+# `ansible-playbook -e "env_prefix=ocp-prod" -e "pv_size=10Gi" -e
+# "cico_project_name=new_fancy_project" adhoc-openshift-pv.yml -i inventory`
+
+
+- hosts: ocp-ci-management
+  become: true
+
+  tasks:
+    - name: Generate a UUID
+      set_fact:
+        pv_uuid: "{{ (cico_project_name + pv_claimref|default('noclaimref', true)) | to_uuid }}"
+
+    - name: UUID Generated
+      debug:
+        var: pv_uuid
+
+    - name: Make a pv name
+      set_fact:
+        pv_name: "pv-{{ pv_size | lower }}-{{ pv_uuid }}"
+
+    - name: UUID Generated
+      debug:
+        var: pv_name
+
+    - name: See if the PV already exists
+      command: "oc get pv/{{ pv_name }}"
+      register: results
+      changed_when: false
+      ignore_errors: true
+
+    - name: Fail - the PV already exists
```

Comment:
Cosmetic but we can have the "fail" task embedded in previous task with "failed_when:" statement